sql-plugin/src/main/scala/com/nvidia/spark/rapids/RmmRapidsRetryIterator.scala

```diff
@@ -794,9 +794,17 @@ object RmmRapidsRetryIterator extends Logging {
   // track the callstack for each memory allocation, don't enable it unless really needed
   val BOOKKEEP_MEMORY_CALLSTACK: Boolean =
     java.lang.Boolean.getBoolean("ai.rapids.memory.bookkeep.callstack")
+  // By default, only print first time to avoid too much log
+  val BOOKKEEP_MEMORY_PRINT_ALL: Boolean =
+    java.lang.Boolean.getBoolean("ai.rapids.memory.bookkeep.printall")
+  var bookkeepPrinted = false
 
   private def logMemoryBookkeeping(): Unit = synchronized { // use synchronized to keep neat
 
+    if (bookkeepPrinted && !BOOKKEEP_MEMORY_PRINT_ALL) {
```

Comment:
```suggestion
    if (!bookkeepPrinted && BOOKKEEP_MEMORY_PRINT_ALL) {
```
then we can wrap everything in that if. That way we omit `return`. 

`return` is a little odd in scala, and can be implemented as an exception. I would avoid if possible.

Reply:
I didn't know the story of `return` in Scala until you mentioned. Just did some chatgpt and learned it, thx! I'll fix this.